src/Config/ConfigLoader.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Robo\Config;
+
+use Grasmash\YamlExpander\Expander;
+
+/**
+ * Load configuration files, and fill in any property values that
+ * need to be expanded.
+ */
+class ConfigLoader implements ConfigLoaderInterface
+{
+    protected $config = [];
+    protected $source = '';
+
+    public function getSourceName()
+    {
+        return $this->source;
+    }
+
+    public function setSourceName($source)
+    {
+        $this->source = $source;
+        return $this;
+    }
+
+    public function import($data)
+    {
+        $this->config = [];
+        return $this->add($data);
+    }
+
+    public function add($data)
+    {
+        if ($data instanceof ConfigLoaderInterface) {
+            $data = $data->export();
+        }
+        $this->config = Expander::expandArrayProperties($data, $this->config);
```

Comment:
This will expand properties using dot notation, but it will not expand dot notated keys. For that, you need something like

```
   $this->config = ArrayManipulator::expandFromDotNotatedKeys($this->config);
   $this->config = Expander::expandArrayProperties($data, $this->config);
   $this->config = ArrayManipulator::expandFromDotNotatedKeys($$this->config);
```